Sources/MapboxMaps/Style/Style+Localization.swift

```diff
@@ -31,41 +31,61 @@ extension Style {
         }
     }
 
-    /// Filters through source to determine supported locale styles.
-    /// This is needed for v7 support
-    internal func getLocaleValue(locale: Locale) -> String? {
-        // Docs for language, region, and script codes  https://developer.apple.com/library/archive/documentation/MacOSX/Conceptual/BPInternational/LanguageandLocaleIDs/LanguageandLocaleIDs.html
-        let supportedLocaleIdentifiers = ["ar", "de", "en", "es", "fr", "it", "ja", "ko", "pt", "ru", "vi", "zh", "zh-Hans", "zh-Hant", "zh-Hant-TW"]
+    /// Returns the BCP 47 language tag supported by Mapbox Streets source v8 that is most preferred according to the given preferences.
+    /// Docs for language, region, and script codes: https://developer.apple.com/library/archive/documentation/MacOSX/Conceptual/BPInternational/LanguageandLocaleIDs/LanguageandLocaleIDs.html
+    internal func preferredMapboxStreetsLocalization(among preferences: [String]) -> String? {
+        let supportedLanguageCodesv8 = ["ar", "en", "es", "fr", "de", "it", "pt", "ru", "zh-Hans", "zh-Hant", "ja", "ko", "vi"].map(Locale.init(identifier:))
+
+        let preferredLocales = preferences.map(Locale.init(identifier:))
+
+        let mostSpecificLanguage = Bundle.preferredLocalizations(from: supportedLanguageCodesv8.map(\.identifier),
+            forPreferences: preferences)
+            .first
 
-        // Do nothing if we do not support the locale
-        if !supportedLocaleIdentifiers.contains(locale.languageCode!) {
+        // `Bundle.preferredLocalizations(from:forPreferences:)` is just returning the first localization it could find.
+        if let mostSpecificLanguage = mostSpecificLanguage,
+            !preferredLocales.contains(where: { $0.languageCode == Locale(identifier: mostSpecificLanguage).languageCode }) {
```

Comment:
I don't like an idea to instantiate Locale with `mostSpecificLanguage` the `N` times (as per `preferredLocales`).
Can you please extract the `Locale(identifier: mostSpecificLanguage).languageCode` part in front of `if` statement.

P.S. Feel free to use [`Optional.map`](https://developer.apple.com/documentation/swift/optional/map(_:)-7txtz) to simplify `Locale` instantiation.

Reply:
👍  -- modified the implementation to use `Optional.map`